stdlib/public/Darwin/Compression/Compression.swift

```diff
@@ -263,32 +323,37 @@ public class InputFilter {
       while _stream.dst_size > 0 && !_endReached {
 
         // Refill _buf if needed, and EOF was not yet read
-        if _stream.src_size == 0 && !_eofReached {
-          _buf = try _readFunc(_bufCapacity) // may be nil
-          // Reset src_size to full _buf size
-          if _buf?.count ?? 0 == 0 { _eofReached = true }
-          _stream.src_size = _buf?.count ?? 0
+        if (_buf?.remaining() ?? 0) == 0 && !_eofReached {
+          let data = try _readFunc(_readCapacity) // may be nil, or empty
+          if data?.count ?? 0 == 0 { // nil or empty -> EOF
+            _eofReached = true
+            _buf = nil
+          } else {
+            _buf = try InputFilterBuffer<D>(data!)
```

Comment:
Minor: `<D>` should not be necessary, I guess it can be inferred.

Reply:
ok